internal/service/resourcepolicy/resource_schema.go

```diff
@@ -0,0 +1,121 @@
+				Computed:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "Unique 24-hexadecimal digit string that identifies an atlas resource policy.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated_by_user": schema.SingleNestedAttribute{
+				Description: "The user that last updated the atlas resource policy.",
+				Computed:    true,
+				Attributes: map[string]schema.Attribute{
+					"id": schema.StringAttribute{
+						Description: "Unique 24-hexadecimal character string that identifies a user.",
+						Computed:    true,
+					},
+					"name": schema.StringAttribute{
+						Description: "Human-readable label that describes a user.",
+						Computed:    true,
+					},
+				},
+			},
+			"last_updated_date": schema.StringAttribute{
+				Description: "Date and time in UTC when the atlas resource policy was last updated.",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "Human-readable label that describes the atlas resource policy.",
+				Optional:    true,
+			},
+			"org_id": schema.StringAttribute{
+				Description: "Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.",
+				Required:    true,
+			},
+			"policies": schema.ListNestedAttribute{
+				Description: "List of policies that make up the atlas resource policy.",
+				Required:    true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"body": schema.StringAttribute{
+							Description: "A string that defines the permissions for the policy. The syntax used is the Cedar Policy language.",
+							Required:    true,
+						},
+						"id": schema.StringAttribute{
+							Description: "Unique 24-hexadecimal character string that identifies the policy.",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"version": schema.StringAttribute{
+				Description: "A string that identifies the version of the atlas resource policy.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+type TFResourcePolicyModel struct {
+	CreatedByUser     types.Object `tfsdk:"created_by_user"`
+	CreatedDate       types.String `tfsdk:"created_date"`
+	ID                types.String `tfsdk:"id"`
+	LastUpdatedByUser types.Object `tfsdk:"last_updated_by_user"`
+	LastUpdatedDate   types.String `tfsdk:"last_updated_date"`
+	Name              types.String `tfsdk:"name"`
+	OrgID             types.String `tfsdk:"org_id"`
+	Policies          types.List   `tfsdk:"policies"`
+	Version           types.String `tfsdk:"version"`
+}
+
+type TFUserMetadataModel struct {
+	ID   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+}
+
+var UserMetadataObjectType = types.ObjectType{AttrTypes: map[string]attr.Type{
```

Comment:
Curious if we can use the TFUserMetadataModel directly or is there preference to use intermediate ObjectType? 

Looking at https://developer.hashicorp.com/terraform/plugin/framework/handling-data/types/object#automatic-conversion-with-tfsdk-struct-tags  I think using model directly should be possible, WDYT?

Reply:
Thanks! Didn't know about this. I'll check it out 👀

Reply:
Discussed offline. We will need to keep it as is.